### ISSUE TITLE
GHC 8.8 tracking ticket

### ISSUE THREAD

**Issue by quasicomputational** [dhall-lang/dhall-haskell#1249]
Tracking ticket for GHC 8.8 issues, mostly dependencies. Updated 2020-05-30:

All packages are compatible with GHC-8.8, but CI doesn't include a GHC-8.8 job yet.

**Comment by jneira** [comment]
I think we are going yo replace yaml with HsYaml soon so we 'll need to track it instead

**Comment by sjakobi** [comment]
#1324 took care of `prettyprinter` and `turtle`, but the other dependencies still need updating.

**Comment by sjakobi** [comment]
I've updated the issue description. We're getting closer! :)

**Comment by quasicomputational** [comment]
Down to 4 blockers now.

**Comment by sjakobi** [comment]
`asn1-encoding` got a compatible release.

Once we have a release that is compatible with GHC-8.8, we should make sure to re-add the Dhall packages to Stackage Nightly, since they were disabled in the recent switch to GHC-8.8.

**Comment by Gabriel439** [comment]
@sjakobi: I already opened a pull request to do so: https://github.com/commercialhaskell/stackage/pull/4860

**Comment by sjakobi** [comment]
I hope that we can soon update CI to use 8.8 instead of 8.4 and 8.6. It might be worth waiting for 8.8.2 and an LTS series using that though.

Regarding `dhall-nix`: `hnix` looks kind of stuck and under-maintained these days. Maybe we should consider switching to the simpler [`language-nix`](https://hackage.haskell.org/package/language-nix) package instead?! Also, does `dhall-nix` have any users at all?!

**Comment by Gabriel439** [comment]
@sjakobi: I'm pretty sure `dhall-nix` has some users, because I got a few complaints when the `dhallToNix` utility in Nixpkgs broke

`language-nix` looks very incomplete to me and I don't think we could use it

**Comment by sjakobi** [comment]
`hnix` finally got a new release, but `hnix-store-core` is still unbuildable with 8.8: https://github.com/haskell-nix/hnix-store/issues/56

**Comment by sjakobi** [comment]
`hnix` finally got a GHC-8.8-compatible release in [v0.8.0](http://hackage.haskell.org/package/hnix-0.8.0). Support for GHC 8.10 is currently WIP: https://github.com/haskell-nix/hnix/issues/592

How about we update CI to use GHC 8.8 for the most recent GHC version?

**Comment by Gabriel439** [comment]
@sjakobi: We would need to upgrade Nixpkgs to a newer version, which is blocked by `static-haskell-nix`

**Comment by sjakobi** [comment]
Thanks @Gabriel439. Are you referring to https://github.com/nh2/static-haskell-nix/issues/73?

**Comment by Gabriel439** [comment]
@sjakobi: Yeah.  That would minimally need to be fixed

**Comment by sjakobi** [comment]
`hnix-0.9` builds with GHC 8.10 too now, and `dhall-nix` too: https://github.com/dhall-lang/dhall-haskell/pull/1862

We can discuss the CI upgrade in a new issue if necessary.
